PySDM/backends/impl_thrust_rtc/methods/collisions_methods.py

```diff
@@ -187,6 +188,198 @@ def __init__(self):
+                "nfmax_is_not_none",
+            ),
+            name_iter="i",
+            body="""
+            frag_size[i] = min(frag_size[i], v_max[i]);
+            frag_size[i] = max(frag_size[i], vmin);
+
+            if (nfmax_is_not_none) {
+                if (x_plus_y[i] / frag_size[i] > nfmax) {
+                    frag_size[i] = x_plus_y[i] / nfmax;
+                }
+            }
+            if (frag_size[i] == 0.0) {
+                frag_size[i] = x_plus_y[i];
+                n_fragment[i] = 1.0;
+            }
+            n_fragment[i] = x_plus_y[i] / frag_size[i];
+            """,
+        )
+
+        self.__gauss_fragmentation_body = trtc.For(
+            param_names=("mu", "sigma", "frag_size", "rand"),
+            name_iter="i",
+            body=f"""
+            frag_size[i] = mu - sigma / {const.sqrt_two} / {const.sqrt_pi} / log(2.) * log(
+                (0.5 + rand[i]) / (1.5 - rand[i])
+            );
+            """,
+        )
+
+        self.__slams_fragmentation_body = trtc.For(
+            param_names=("n_fragment", "frag_size", "x_plus_y", "probs", "rand"),
+            name_iter="i",
+            body="""
+            probs[i] = 0.0;
+            n_fragment[i] = 1;
+
+            for (auto n = 0; n < 22; n += 1) {
+                probs[i] += 0.91 * pow((n + 2), (-1.56));
+                if (rand[i] < probs[i]) {
+                    n_fragment[i] = n + 2;
+                    break;
+                }
+            }
+            frag_size[i] = x_plus_y[i] / n_fragment[i];
+            """,
+        )
+
+        self.__feingold1988_fragmentation_body = trtc.For(
+            param_names=("scale", "frag_size", "x_plus_y", "rand", "fragtol"),
+            name_iter="i",
+            body="""
+            auto log_arg = max(1 - rand[i] * scale / x_plus_y[i], fragtol);
+            frag_size[i] = -scale * log(log_arg);
+            """,
+        )
+
+        self.__straub_Nr_body = """
+
+            if (gam[i] * CW[i] >= 7.0) {
+                Nr1[i] = 0.088 * (gam[i] * CW[i] - 7.0);
+            }
+            if (CW[i] >= 21.0) {
+                Nr2[i] = 0.22 * (CW[i] - 21.0);
+                if (CW[i] <= 46.0) {
+                    Nr3[i] = 0.04 * (46.0 - CW[i]);
+                }
+            }
+            else {
+                Nr3[i] = 1.0;
+            }
+            Nr4[i] = 1.0;
+            Nrt[i] = Nr1[i] + Nr2[i] + Nr3[i] + Nr4[i];
+        """
+
+        self.__straub_p1 = f"""
+
+                auto E_D1 = 0.04 * CM;
+                auto delD1 = 0.125 * pow(CW[i], 0.5);
+                auto var_1 = pow(delD1, 2. / 12.);
+                auto sigma1 = sqrt(log(var_1 / pow(E_D1, 2) + 1));
+                auto mu1 = log(E_D1) - pow(sigma1, 2) / 2.;
+                auto X = rand[i];
+
+                frag_size[i] = exp(
+                    mu1
+                    - sigma1 / {const.sqrt_two} / {const.sqrt_pi} / log(2.) * log((0.5 + X) / (1.5 - X))
+                );
```

Comment:
Could consider (in both backends) creating a separate function call for this approximation of a gaussian CDF since it is used in multiple places

Reply:
best if added to `Formulae.Trivia` - then, we can use it from both CPU and GPU backends without coding twice